File: sample/SphereSample/SphereSample/SampleWebViewController.m
@import WebKit;
@import SphereSDK;

#import "SampleWebViewController.h"

@interface SampleWebViewController () <WKScriptMessageHandler>
@property (nonatomic, readonly) WKWebView *webView;
@end

@implementation SampleWebViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    // Initialize the webview and add self as a script message handler.
    WKWebViewConfiguration *configuration = [[WKWebViewConfiguration alloc] init];
    WKUserContentController *controller = [[WKUserContentController alloc] init];

    [controller addScriptMessageHandler:self name:@"sphere"];
    configuration.userContentController = controller;
    _webView = [[WKWebView alloc] initWithFrame:[UIScreen mainScreen].bounds configuration:configuration];
    [self.view addSubview:self.webView];

    // Navigate to site
    NSString* fileURL = [[[NSBundle mainBundle] resourcePath] stringByAppendingPathComponent:@"web/index.html"];
    NSURLRequest *request = [NSURLRequest requestWithURL:[NSURL fileURLWithPath:fileURL]];
//    NSURLRequest *request = [NSURLRequest requestWithURL:[NSURL URLWithString:@"your website url"]];

    [self.webView loadRequest:request];
}

- (void)userContentController:(WKUserContentController *)userContentController
      didReceiveScriptMessage:(WKScriptMessage *)message {

    [SPRScriptMessageHandler handleReceiveScriptMessage:message];
}
@end

File: sample/SphereSample/SphereSampleSwift/SampleWebViewController.swift
import UIKit
import WebKit
import SphereSDK

class SampleWebViewController: UIViewController, WKScriptMessageHandler {
    private var webView: WKWebView!

    override func viewDidLoad() {
        super.viewDidLoad()

        // Initialize the webview and add self as a script message handler.
        self.webView = WKWebView(frame: self.view.frame)
        self.webView.configuration.userContentController.add(self, name: "sphere")
        self.view.addSubview(self.webView)

        // Navigate to site
        guard let url = Bundle.main.url(forResource: "index", withExtension: "html", subdirectory: "web") else { return }

        self.webView.loadFileURL(url, allowingReadAccessTo: url)

//        let request = URLRequest(url: URL(string: "your website url")!)
//        self.webView.load(request)
    }

    func userContentController(_ userContentController: WKUserContentController,
                               didReceive message: WKScriptMessage) {

        SPRScriptMessageHandler.handleReceive(message)
    }
}
